pupil_src/shared_modules/math_helper/transformations.py

```diff
@@ -1980,6 +1983,33 @@ def _import_module(name, package=None, warn=True, prefix="_py_", ignore="_"):
         return True
 
 
+def find_rigid_transform(A, B):
+    # we expect the shape to be of length 2
+    assert len(A.shape) == len(B.shape) == 2
```

Comment:
Assertions should generally not be used to make sure function parameters are given the correct arguments. Wrong parameters should throw e.g. a `ValueError` or in this case numpy should anyway throw an error in line 1999 because arrays of different lengths can not be multiplied like that. 
Please check e.g. [this](https://github.com/emre/notes/blob/master/python/when-to-use-assert.md) resource on when to use assertions and evaluate if they make sense at this point.

Reply:
True fact. I will finish the rest of the calibration code first and put it on the list of things to do. This was in the code since ages and I just moved it to a more appropriate place. But I agree, this function needs some work.